clients/src/main/java/org/apache/kafka/common/network/SSLTransportLayer.java

```diff
@@ -222,11 +222,13 @@ public void handshake() throws IOException {
                               channelId, appReadBuffer.position(), netReadBuffer.position(), netWriteBuffer.position());
                     handshakeResult = handshakeWrap(write);
                     if (handshakeResult.getStatus() == Status.BUFFER_OVERFLOW) {
-                        int currentPacketBufferSize = packetBufferSize();
-                        netWriteBuffer = Utils.ensureCapacity(netWriteBuffer, currentPacketBufferSize);
-                        if (netWriteBuffer.position() >= currentPacketBufferSize) {
+                        int currentNetWriteBufferSize = netWriteBufferSize();
+                        netWriteBuffer.position(netWriteBuffer.limit());
```

Comment:
It may be difficult to show the need for `position()` in a test, but I will try.
The scenario is: 
1. `wrap()` completes with some data in the buffer (eg. position=15)
2. `handshakeWrap()` flips the buffer (limit=15, position=0)
3. `Utils.ensureCapacity()` flips the buffer again (limit=0, position=0) 

If `position` is not reset to `limit` before `Utils.ensureCapacity()` is invoked, the value in `limit` which shows how much data is in the buffer would be lost. And `limit` and `position` of the resulting buffer would be zero. And hence the data in the existing buffer would not get added to the new buffer.